asdf_compression/blsc/compressor.py

```diff
@@ -7,10 +7,34 @@ class BloscCompressor(Compressor):
     def compress(self, data, **kwargs):
         import blosc
 
-        yield blosc.compress(data, **kwargs)
+        # Type size, necessary for shuffle filters
+        typesize = data.itemsize
+
+        # Compression level
+        clevel = kwargs.get("level", 9)
+
+        # Shuffle filter
+        shuffle_string = kwargs.get("filter", "bitshuffle")
+        if shuffle_string == "noshuffle":
+            shuffle = blosc.NOSHUFFLE
+        elif shuffle_string == "byteshuffle":
+            shuffle = blosc.BYTESHUFFLE
+        else:
+            # Fallback and default
+            shuffle = blosc.BITSHUFFLE
+
+        # CODEC name
+        cname = kwargs.get("codec", "blosclz")
+
+        # Number of threads
+        nthreads = kwargs.get("threads", 1)
+        self._api.set_nthreads(nthreads)
```

Comment:
Should this be:
```suggestion
        blosc.set_nthreads(nthreads)
```
If so, I'd say it's cleaner to leave this up to the user to call prior to saving or opening an asdf file and leave it out of the api for this compressor.

Reply:
Putting it into the compressor allows people to use the compressor without themselves depending on the `blosc` package.

Reply:
That's a good point that it would mean that a user could just pass in 'nthreads' to the compression kwargs and not need to call blosc.

Does `set_nthreads` also effect decompression? Looking at how asdf handles decompression there is no mechanism for providing keyword arguments even though the Compressor supports them (that is certainly a bug and gap in the api).

Reply:
I think decompression is also multi-threaded. I'm not sure.